packages/blocks/src/block-content-provider/index.js

```diff
@@ -23,28 +30,43 @@ import { serialize } from '../api';
  * </BlockContentProvider>
  * ```
  */
-class BlockContentProvider extends Component {
-	getChildContext() {
-		const { innerBlocks } = this.props;
+const BlockContentProvider = ( { children, innerBlocks } ) => {
+	const BlockContent = () => {
+		// Value is an array of blocks, so defer to block serializer
+		const html = serialize( innerBlocks );
 
-		return {
-			BlockContent() {
-				// Value is an array of blocks, so defer to block serializer
-				const html = serialize( innerBlocks );
+		// Use special-cased raw HTML tag to avoid default escaping
+		return <RawHTML>{ html }</RawHTML>;
+	};
 
-				// Use special-cased raw HTML tag to avoid default escaping
-				return <RawHTML>{ html }</RawHTML>;
-			},
-		};
-	}
-
-	render() {
-		return this.props.children;
-	}
-}
-
-BlockContentProvider.childContextTypes = {
-	BlockContent: () => {},
+	return (
+		<Provider value={ { BlockContent } }>
+			{ children }
+		</Provider>
+	);
 };
 
+/**
+ * A Higher Order Component used to inject BlockContent context to the
+ * wrapped component.
+ *
+ * @param {Function} mapContextToProps Function called on every context change,
+ *                                     expected to return object of props to
+ *                                     merge with the component's own props.
+ *
+ * @return {Component} Enhanced component with injected context as props.
+ */
+export const withBlockContentContext = ( mapContextToProps ) => createHigherOrderComponent( ( OriginalComponent ) => {
```

Comment:
To my previous point, I don't see why we'd be passing anything other than the content component. We shouldn't need to map anything.

Reply:
I did it for consistency with other context related HOCs. However, you made a good point. I will refactor as suggested.